docs/excel/excel-add-ins-ranges-advanced.md

```diff
@@ -0,0 +1,166 @@
+- `"Values"` copies the data values and, in the case of formulas, the result of the formula.
+- `"Formats"` copies the formatting of the range, including font, color, and other format settings, but no values.
+- `"All"` (the default option) copies both data and formatting, preserving cells’ formulas if found.
+
+`skipBlanks` sets whether blank cells are copied into the destination. When true, `copyFrom` skips blank cells in the source range. 
+Skipped cells will not overwrite the existing data of their corresponding cells in the destination range. The default is false.
+
+The following code sample and images demonstrate this behavior in a simple scenario.
+
+```js
+Excel.run(function (context) {
+    var sheet = context.workbook.worksheets.getItem("Sample");
+    // copy a range, omitting the blank cells so existing data is not overwritten in those cells
+    sheet.getRange("D1").copyFrom("A1:C1",
+        Excel.RangeCopyType.all,
+        true, // skipBlanks
+        false); // transpose
+    // copy a range, including the blank cells which will overwrite existing data in the target cells
+    sheet.getRange("D2").copyFrom("A2:C2",
+        Excel.RangeCopyType.all,
+        false, // skipBlanks
+        false); // transpose
+    return context.sync();
+}).catch(errorHandlerFunction);
+```
+
+*Before the preceding function has been run.*
+
+![Data in Excel before range’s copy method has been run](../images/excel-range-copyfrom-skipblanks-before.png)
+
+*After the preceding function has been run.*
+
+![Data in Excel after range’s copy method has been run](../images/excel-range-copyfrom-skipblanks-after.png)
+
+`transpose` determines whether or not the data is transposed, meaning its rows and columns are switched, into the source location.
+A transposed range is flipped along the main diagonal, so rows **1**, **2**, and **3** will become columns **A**, **B**, and **C**.
+
+## Remove duplicates
+
+> [!NOTE]
+> The Range object's `removeDuplicates` function is currently available only in public preview (beta). To use this feature, you must use the beta library of the Office.js CDN: https://appsforoffice.microsoft.com/lib/beta/hosted/office.js.
+> If you are using TypeScript or your code editor uses TypeScript type definition files for IntelliSense, use https://appsforoffice.microsoft.com/lib/beta/hosted/office.d.ts.
+
+The Range object's `removeDuplicates` function removes rows with duplicate entries in the specified columns. The function goes through each row in the range from the lowest-valued index to the highest-valued index in the range (from top to bottom). A row is deleted if it has a value in the specified column or columns appeared earlier in the range. Rows in the range below the deleted row are shifted up. `removeDuplicates` does not affect the position of cells outside of the range.
```

Comment:
The third sentence is worded awkwardly...perhaps remove the words "it has"?

-> "A row is deleted if a value in its specified column or columns appeared earlier in the range."